tests/std/tests/P0896R4_ranges_alg_reverse/test.cpp

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <ranges>
+#include <utility>
+
+#include <range_algorithm_support.hpp>
+
+using namespace std;
+
+// Validate dangling story
+STATIC_ASSERT(same_as<decltype(ranges::reverse(borrowed<false>{})), ranges::dangling>);
+STATIC_ASSERT(same_as<decltype(ranges::reverse(borrowed<true>{})), int*>);
+
+struct instantiator {
+    static constexpr int expected[] = {1367, 42, 13};
+
+    template <class Bidi>

```

Comment:
I think this would be clearer as `ranged::bidirectional_range ReadWrite`

Reply:
I've decided it's silly to give the template type parameter a convoluted name of we're going to constrain it and simply used `ranges::bidirectional_range R`.